torchci/clickhouse_queries/master_commit_red/query.sql

```diff
@@ -1,38 +1,56 @@
 --- This query is used to show the histogram of trunk red commits on HUD metrics page
 --- during a period of time
-WITH all_jobs AS (
+-- Split up the query into multiple CTEs to make it faster.
+with commits as (
+  select
+    push.head_commit.'timestamp' as time,
+    push.head_commit.'id' as sha
+  from
+    push final
+  where
+    push.ref in ('refs/heads/master', 'refs/heads/main')
+    and push.repository.'owner'.'name' = 'pytorch'
+    and push.repository.'name' = 'pytorch'
+    and push.head_commit.'timestamp' >= {startTime: DateTime64(3)}
+    and push.head_commit.'timestamp' < {stopTime: DateTime64(3)}
+),
+all_runs as (
+  select
+    workflow_run.id as id,
+    workflow_run.head_commit.'id' as sha,
+    workflow_run.name as name,
+    commit.time as time
+  from
+    workflow_run final
+    join commits commit on workflow_run.head_commit.'id' = commit.sha
+  where
+    (
+      -- Limit it to workflows which block viable/strict upgrades
+      workflow_run.name in ('Lint', 'pull', 'trunk')
+      OR workflow_run.name like 'linux-binary%'
+    )
+    AND workflow_run.event != 'workflow_run' -- Filter out workflow_run-triggered jobs, which have nothing to do with the SHA
+    and workflow_run.id in (select id from materialized_views.workflow_run_by_head_sha where head_sha in (select sha from commits))
+),
+all_jobs AS (
   SELECT
-    push.head_commit.'timestamp' AS time,
+    workflow_run.time AS time,
     CASE
       WHEN job.conclusion = 'failure' THEN 'red'
       WHEN job.conclusion = 'timed_out' THEN 'red'
       WHEN job.conclusion = 'cancelled' THEN 'red'
       WHEN job.conclusion = '' THEN 'pending'
       ELSE 'green'
     END as conclusion,
-    push.head_commit.'id' AS sha
+    workflow_run.sha AS sha
   FROM
-    workflow_job job FINAL
-    JOIN workflow_run FINAL ON workflow_run.id = workflow_job.run_id
-    JOIN push FINAL ON workflow_run.head_commit.'id' = push.head_commit.'id'
+    default.workflow_job job final join all_runs workflow_run on workflow_run.id = workflow_job.run_id
```

Comment:
This reads a bit confusing IMO.  If I read it correctly, the `all_runs` table has the alias as `workflow_run`, which has the correct syntax.  But I always think of `workflow_run` as the `workflow_run` table instead of having it as an alias